[Bugfix] Force portrait mode in setup view

cheukchuen opened the pull request:
## Purpose
<!-- Describe the intention of the changes being proposed. What problem does it solve or functionality does it add? -->
* Only stop receiving orientation notification in setup view
  * As a temporary workaround for local video stream to not rotate the video when rotating screen
* Calling view still receives orientation notification while orientation locked when displaying drawer
  * So that screen rotates to proper orientation when dismissing drawer for participants list/audio device selection

## Does this introduce a breaking change?
<!-- Mark one with an "x". -->
```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What to Check
Verify that the following are valid
* Setup view remains in portrait mode when screen was in landscape mode before entering setup screen
* Setup view remains in portrait mode when screen was in landscape mode during the call and after call has ended
* Calling view able to rotate to proper orientation after dismissing the drawer
  * Able to rotate to portrait mode after dismissing the drawer in landscape mode
  * Able to rotate to landscape mode after dismissing the drawer in portrait mode

palatter reviewed (dismissed):
Looks good.

Tested rotation with background/foreground, video and screen share

JoshuaLai reviewed (dismissed):
Tested and its working as expected. But I was never able to reproduce this issue so I would be curious @AmyL219 if this fixes it for you.

cheukchuen committed:
Force portrait mode in setup view

- Only stop receiving orientation notification in setup view
- Calling view still receives orientation notification while orientation locked when displaying drawer

cheukchuen committed:
Update CHANGELOG.md

JoshuaLai reviewed (dismissed):
Looks good still the same changes

cheukchuen committed:
Stop listening to orientation on first load of SetupView

jimchou-dev reviewed (dismissed):
LGTM

AmyL219 reviewed (dismissed):
Test. Couldn't reproduce the bug in this morning! Looks good

AmyL219 commented on AzureCommunicationUI/AzureCommunicationUI/Presentation/SwiftUI/Container/ContainerUIHostingController.swift:
```diff
@@ -49,37 +51,36 @@ class ContainerUIHostingController: UIHostingController<ContainerUIHostingContro
             }).store(in: cancelBag)
     }
 
+    private func haltSetupViewOrientation(containerView: ContainerView) {
```
Why do we need this in the initilizer? Would this be duplicated to line 60 -62 though?

cheukchuen replied:
Need this in `init` for when the composite first loads. `supportedOrientations` is set to `portrait` initially, so couldn't reach lines 60-62 due to `.removeDuplicates()`.

Of course we could also remove `.removeDuplicates()` and `haltSetupViewOrientation(ContainerView:)`, but this hides the issue and in the future someone else might unknowingly add `.removeDuplicates()` back

AmyL219 replied:
Okey, let keep them both there

AmyL219 commented on AzureCommunicationUI/AzureCommunicationUI/Presentation/SwiftUI/Container/ContainerUIHostingController.swift:
```diff
@@ -49,37 +51,36 @@ class ContainerUIHostingController: UIHostingController<ContainerUIHostingContro
+                if containerView.router.currentView == .setupView,
+                   UIDevice.current.isGeneratingDeviceOrientationNotifications {
+                    UIDevice.current.endGeneratingDeviceOrientationNotifications()
+                } else if !UIDevice.current.isGeneratingDeviceOrientationNotifications {
```
I feel like the workaround Kevin implemented before doesn't work anymore. The local videostream keeps rotating inside the setup view. Could @kevinyulu you have test on PR? 
Either we remove the workaround completely or fix the workaround.

kevinyulu replied:
@AmyL219  yes I tested and confirmed that the local video camera scene orientation "flipped" when device is rotated.

cheukchuen replied:
Latest commit fixes the issue, please try again thanks!

kevinyulu replied:
@cheukchuen  how about we add one check in the else if above? 

```
} else if containerView.router.currentView != .setupView,
!UIDevice.current.isGeneratingDeviceOrientationNotifications {
UIDevice.current.beginGeneratingDeviceOrientationNotifications()
 ```
 
 I tested this and it looked like it does the following:
 * stop camera orientation accidentally flipped when device is rotated in setup view, and
 * allow orientation update after participant drawer is dismissed. 
 Let me know if there is other side effect this approach introduces.

cheukchuen replied:
That's correct @kevinyulu, in the setup view we want to force portrait mode and prevent camera orientation flipping, whereas for calling view we want the screen to be rotated after drawer dismissal at the expense of camera flipping.

Good idea on the suggestion as well. The initial flow condition we had for this PR resulted in the `else` condition being evaluated for setup view, hence the issue reported by @AmyL219. Made a change to use `switch` to be specific about what should occur for each view.

cheukchuen committed:
Fix halt setupview orientation condition

cheukchuen committed:
Merge branch 'develop/beta' into bug/screen-rotation

cheukchuen committed:
Improve readability with switch statement

cheukchuen committed:
Merge branch 'develop/beta' into bug/screen-rotation

AmyL219 committed:
Merge branch 'develop/beta' into bug/screen-rotation
